tools/toolchains/__init__.py

```diff
@@ -107,34 +107,38 @@ def __radd__(self, resources):
         else:
             return self.add(resources)
 
-    def add(self, resources):
-        for f,p in resources.file_basepath.items():
-            self.file_basepath[f] = p
-
+    def add_not_source(self, resources):
```

Comment:
What is add_not_source()? Are these changes needed only to support specific test routine??